test/new-e2e/utils/infra/stack_manager.go

```diff
@@ -113,6 +114,17 @@ func (sm *StackManager) GetStack(ctx context.Context, name string, config runner
 		FlowToPlugins: true,
 		LogLevel:      &loglevel,
 	}))
+
+	if err != nil {
+		_, errDestroy := stack.Destroy(ctx, optdestroy.ProgressStreams(os.Stderr), optdestroy.DebugLogging(debug.LoggingOptions{
```

Comment:
What do you think about using `deleteStack` instead?

Reply:
Yes I can do this, I could not use `DeleteStack` because of locks, but `deleteStack should work fine